### PR TITLE
Added Strato as Provider

### PR DESCRIPTION
**Pull request by Phiwatec**
I added the german hosting company Strato as a provider.
It is basically the same as google with the url and login params changed.

### CODE REVIEW

**Comment by TimothyYe** [comment]
Hi @Phiwatec  , I see a new file added `internal/provider/hetzner/hetzner_handler.go` but it's empty. May I know what this file is used for?

**Comment by Phiwatec** [comment]
> Hi @Phiwatec , I see a new file added `internal/provider/hetzner/hetzner_handler.go` but it's empty. May I know what this file is used for?

Sorry I messed up that file. Was supposed to be on a different branch. I now deleted the empty file and folder. :)

**Comment by TimothyYe** [comment]
> > Hi @Phiwatec , I see a new file added `internal/provider/hetzner/hetzner_handler.go` but it's empty. May I know what this file is used for?
> 
> Sorry I messed up that file. Was supposed to be on a different branch. I now deleted the empty file and folder. :)

I see, thanks.

BTW, to register a new DNS provider, I think you should also update the file `factory.go`, and pay attention to the function  `GetProvider`, all the providers are registered here. So did you test this newly added provider? Does it work well?

**Comment by Phiwatec** [comment]
It works now. I updated all necessary files. (Took a bit. Was on vacation :) )
I added one line in google_handler because the system did not handle a wrong password or domain name.
Google and strato return either badauth or nohost when auth failed or the domain to update is unknown respectively.
Now it checks for 'good' or 'nochg' and returns  an error otherwise.

File: internal/provider/google/google_handler.go [hunk 1]
```diff
@@ -61,13 +61,15 @@ func (provider *DNSProvider) updateIP(domain, subDomain, currentIP string) error
 	body, _ := io.ReadAll(resp.Body)
 	if resp.StatusCode != http.StatusOK {
 		log.Errorf("Update IP failed: %s", string(body))
-		return fmt.Errorf("update IP failed: %s", string(body))
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
Comment (TimothyYe) [inline internal/provider/google/google_handler.go:64]: In Go, error messages are usually written in lower-case.
Please refer to: https://earthly.dev/blog/golang-errors/

File: internal/provider/google/google_handler.go [hunk 2]
```diff
@@ -61,13 +61,15 @@ func (provider *DNSProvider) updateIP(domain, subDomain, currentIP string) error
...
 	}
 
 	if strings.Contains(string(body), "good") {
 		log.Infof("Update IP success: %s", string(body))
 	} else if strings.Contains(string(body), "nochg") {
 		log.Infof("IP not changed: %s", string(body))
+	} else {
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
Comment (TimothyYe) [inline internal/provider/google/google_handler.go:72]: Same here

File: internal/provider/strato/strato_handler.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+	if err != nil {
+		log.Error("Err:", err.Error())
+		return err
+	}
+
+	body, _ := io.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("Update IP failed: %s", string(body))
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
Comment (TimothyYe) [inline internal/provider/strato/strato_handler.go:64]: Please use lower-case for the error message.

File: internal/provider/strato/strato_handler.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	if strings.Contains(string(body), "good") {
+		log.Infof("Update IP success: %s", string(body))
+	} else if strings.Contains(string(body), "nochg") {
+		log.Infof("IP not changed: %s", string(body))
+	} else {
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
Comment (TimothyYe) [inline internal/provider/strato/strato_handler.go:72]: Please use lower-case for the error message.

File: internal/provider/strato/strato_handler.go (hunk 3 above)
Comment (TimothyYe) [reply internal/provider/strato/strato_handler.go:64]: Oh, I mean the line 64, not the line 63

Comment (Phiwatec) [reply internal/provider/strato/strato_handler.go]: Sorry my bad. Edited the wrong line

**Comment by TimothyYe** [comment]
@Phiwatec Thank you for your contribution!

**Pull request merged**